src/Http/Routing/src/RouteOptions.cs

```diff
@@ -125,11 +125,12 @@ internal ICollection<EndpointDataSource> EndpointDataSources
         return defaults;
     }
 
-   /// <summary>
-   /// Adds or overwrites the parameter policy with the associated route pattern token.
-   /// </summary>
+    /// <summary>
+    /// Adds or overwrites the parameter policy with the associated route pattern token.
+    /// </summary>
     /// <typeparam name="T">The parameter policy type.</typeparam>
     /// <param name="token">The route token used to apply the parameter policy.</param>
+    [UnconditionalSuppressMessage("Trimming", "IL2026:RequiresUnreferencedCodeAttribute", Justification = "DynamicallyAccessedMembers preserves type")]
```

Comment:
Why does this complain?

Reply:
ConstraintMap has an attribute on it to warn people that it uses unreferenced code, and this method uses ConstraintMap.

Reply:
Ah! I added that attribute LOL

Reply:
A simple workaround is to use the field _constraintTypeMap here and in other places.